### PR TITLE
Adding prometheus metrics to remaining modules

### PR DESCRIPTION
**Pull request by samincheva**
## Changes proposed in this PR
- Adds prometheus metrics to match any remaining codahale metrics.

## Why are we making these changes?
Ongoing migration to prometheus.

### CODE REVIEW

File: scheduler/src/cook/mesos/sandbox.clj [hunk 1]
```diff
@@ -129,6 +139,7 @@
   [{:keys [pending-sync-hosts] :as pending-sync-state}]
   (counters/clear! sandbox-pending-sync-host-count)
   (when (seq pending-sync-hosts)
+    (prom/set prom/mesos-pending-sync-host-count (count pending-sync-hosts))
```
Comment (scrosby) [inline scheduler/src/cook/mesos/sandbox.clj:142]: Should this be inc?

File: scheduler/src/cook/progress.clj [hunk 2]
```diff
@@ -57,13 +59,16 @@
...
+            (prom/set prom/progress-aggregator-pending-states-count 0)
             (counters/clear! progress-aggregator-pending-states-count))
+          (prom/inc prom/progress-aggregator-pending-states-count (- new-count old-count))
           (counters/inc! progress-aggregator-pending-states-count (- new-count old-count)))
         instance-id->progress-state')
       (do
         (log/warn "skipping" data "as it is missing an integer progress-sequence")
         instance-id->progress-state))
     (do
+      (prom/inc prom/progress-aggregator-drop-count)
```
Comment (scrosby) [inline scheduler/src/cook/progress.clj:71]: Does this replace both the rate and the count?

Comment (samincheva) [reply scheduler/src/cook/progress.clj:71]: Yes, we can use the counter variable with a `rate` function to calculate the rate dynamically.

File: scheduler/src/cook/mesos/sandbox.clj (hunk 1 above)
Comment (samincheva) [reply scheduler/src/cook/mesos/sandbox.clj:143]: I was missing the call to set the counter value to 0 (like we do for the codahale counter with the call to clear!). Updated to have the call to set to 0, and then it's functionally equivalent to set or inc by that amount.

**Pull request merged**